editions/dev/tiddlers/new/Hook_ th-before-importing.tid

```diff
@@ -0,0 +1,62 @@
+created: 20210203184649726
+modified: 20210203191622629
+tags: HookMechanism
+title: Hook: th-before-importing
+type: text/vnd.tiddlywiki
+
+This hook allows plugins to inspect or modify the `importTiddler` object ''before'' any tiddlers are imported. It is invoked when the final "Import" button is clicked, but ''before'' the selected tiddlers are being imported into the store.
```

Comment:
Have you checked that modifying the importTiddler object is actually reflected in the tiddlers imported? I suspect this is not the case, as the importData variable has already been assigned.

Reply:
>Have you checked that modifying the importTiddler object is actually reflected in the tiddlers imported?

Yes! The bundler also contains some upgraders, that are executed earlier in `NavigatorWidget.prototype.handleImportTiddlersEvent`, which makes it possible to manipulate the importTiddler data in this hook. 

At the moment I'm only using the hook, to write a "header" line into a log-tiddler. This doesn't need to manipulate the importTiddler data.

Reply:
This is incorrect and the docs are misleading. The documentation suggests that the `importTiddler` can be freely modified and the changes will be reflected in the imported tiddlers. However, any changes to the `JSON` payload are not reflected in the imported tiddlers.